pinecone/data/vector_factory.py

```diff
@@ -0,0 +1,109 @@
+import numbers
```

Comment:
This began as a copy/paste of the removed code in `Index` but I refactored it a good bit to cover some cases that weren't being handled.